web/client/utils/SecurityUtils.js

```diff
@@ -6,18 +6,196 @@
+        // getting the user attributes
+        let userAttributes = this.getUserAttributes();
+        if (!userAttributes || !attributeName ) {
+            // the user as no attributes or the provided attribute name is undefined
+            return undefined;
+        }
+        return userAttributes.find(attribute => attribute.name
+            && attribute.name.toLowerCase() === attributeName.toLowerCase());
+    },
+
+    /**
+     * Search in the user attributes an attribute that matchs the provided
+     * attribute name. The search will not be case sensitive. Undefined is
+     * returned if the attribute could not be found otherwise the attribute
+     * value is returned.
+     */
+    findUserAttributeValue: function(attributeName) {
+        let userAttribute = this.findUserAttribute(attributeName);
+        return userAttribute && userAttribute.value;
+    },
+
+    /**
+     * Returns an array with the configured authentication rules. If no rules
+     * were configured an empty array is returned.
+     */
+    getAuthenticationRules: function() {
+        return ConfigUtils.getConfigProp('authenticationRules') || [];
+    },
+
+    /**
+     * Checks if authentication is activated or not.
+     */
+    isAuthenticationActivated: function() {
+        return ConfigUtils.getConfigProp('useAuthenticationRules') || false;
+    },
+
+    /**
+     * Returns the authentication method that should be used for the provided URL.
+     * We go through the authentication rules and find the first one that matchs
+     * the provided URL, if no rule matchs the provided URL undefined is returned.
+     */
+    getAuthenticationMethod: function(url) {
+        const foundRule = this.getAuthenticationRules().find(
+            rule => rule && rule.urlPattern && url.match(new RegExp(rule.urlPattern, "i")));
+        return foundRule ? foundRule.method : undefined;
+    },
+
+    /**
+     * This method will add to the axios config object the correct authentication
+     * method based on the request URL.
+     */
+    addAuthenticationToAxios: function(axiosConfig) {
+        if (!axiosConfig || !axiosConfig.url || !this.isAuthenticationActivated()) {
+            return axiosConfig;
+        }
+        switch (this.getAuthenticationMethod(axiosConfig.url)) {
+            case 'authkey':
+                const userUuid = this.findUserAttributeValue('uuid');
+                if (!userUuid) {
+                    return axiosConfig;
+                }
+                addParameterToAxiosConfig(axiosConfig, 'authkey', userUuid);
+                return axiosConfig;
+            case 'basic':
+                const userName = this.getUserName();
+                const userPassword = this.getUserPassword();
+                if (!userName || !userPassword) {
+                    return axiosConfig;
+                }
+                // axios support basic authentication natively
+                axiosConfig.auth = {
```

Comment:
Ignore auth object and inject the header instead. Something like:  
https://github.com/geosolutions-it/MapStore2/blob/master/web/client/api/GeoStoreDAO.js#L65